Move measurevariable

Wenxing-hou opened the pull request.

Wenxing-hou committed:
SecurityPkg: Add PcdEnableSpdmDeviceAuhenticaion and check

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

Wenxing-hou committed:
SecurityPkg/Tcg2Dxe: Add EV_EFI_SPDM_DEVICE_POLICY measure

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

jyao1 commented on SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c:
```diff
@@ -2333,6 +2334,36 @@ MeasureAllSecureVariables (
   }
 
+  //
+  // Meaurement UEFI device signature database
```
Please check AuthLib as well.

Wenxing-hou replied:
Thanks. I have added the measure.

Wenxing-hou committed:
MdeModulePkg:Add PcdEnableSpdmDeviceAuhenticaion and variable udpate measure

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

Wenxing-hou committed:
MdePkg: Move gEdkiiDeviceSignatureDatabaseGuid to MdePkg

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

jyao1 commented on MdePkg/MdePkg.dec:
```diff
@@ -739,6 +739,12 @@
+  #
+  ## GUID used to specify section with devdb content
+  ## Include/Guid/DeviceAuthentication.h
+  gEdkiiDeviceSignatureDatabaseGuid  = {0xb9c2b4f4, 0xbf5f, 0x462d, {0x8a, 0xdf, 0xc5, 0xc7, 0xa, 0xc3, 0x5d, 0xad}}
```
why use Edkii as prefix?

Wenxing-hou replied:
I have fixed the name.

jyao1 commented on MdeModulePkg/Universal/Variable/RuntimeDxe/Measurement.c:
```diff
@@ -32,6 +33,7 @@ VARIABLE_TYPE  mVariableType[] = {
   { EFI_IMAGE_SECURITY_DATABASE,  &gEfiImageSecurityDatabaseGuid },
   { EFI_IMAGE_SECURITY_DATABASE1, &gEfiImageSecurityDatabaseGuid },
   { EFI_IMAGE_SECURITY_DATABASE2, &gEfiImageSecurityDatabaseGuid },
+{ EDKII_DEVICE_SECURITY_DATABASE, &gEdkiiDeviceSignatureDatabaseGuid },
```
why use EDKII prefix?

Wenxing-hou committed:
Change prefix

Change gEdkiiDeviceSignatureDatabaseGuid and
EDKII_DEVICE_SECURITY_DATABASE prefix

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

Zhiqiang520 commented on SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c:
```diff
@@ -2333,6 +2334,36 @@ MeasureAllSecureVariables (
+      Status = MeasureVariable (
+                PCR_INDEX_FOR_SIGNATURE_DB,
+                EV_EFI_SPDM_DEVICE_POLICY,
+                EDKII_DEVICE_SECURITY_DATABASE,
```
I think here `EDKII_DEVICE_SECURITY_DATABASE` should be changed to `EFI_DEVICE_SECURITY_DATABASE`.

Wenxing-hou replied:
Thanks. I have fixed the name error.

Zhiqiang520 commented on SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c:
```diff
@@ -2333,6 +2334,36 @@ MeasureAllSecureVariables (
+      Status = MeasureVariable (
+                PCR_INDEX_FOR_SIGNATURE_DB,
+                EV_EFI_SPDM_DEVICE_POLICY,
+                EDKII_DEVICE_SECURITY_DATABASE,
```
I think here `EDKII_DEVICE_SECURITY_DATABASE` should be changed to `EFI_DEVICE_SECURITY_DATABASE`.
